torch/nn/modules/conv.py

```diff
@@ -344,31 +344,25 @@ def forward(self, input, output_size=None):
 
     def _output_padding(self, input, output_size):
         if output_size is None:
-            return self.output_padding
+            return self.padding, self.output_padding
 
         output_size = list(output_size)
         k = input.dim() - 2
         if len(output_size) == k + 2:
-            output_size = output_size[-2:]
+            output_size = output_size[2:]
         if len(output_size) != k:
             raise ValueError(
                 "output_size must have {} or {} elements (got {})"
                 .format(k, k + 2, len(output_size)))
 
-        def dim_size(d):
-            return ((input.size(d + 2) - 1) * self.stride[d] -
-                    2 * self.padding[d] + self.kernel_size[d])
-
-        min_sizes = [dim_size(d) for d in range(k)]
-        max_sizes = [min_sizes[d] + self.stride[d] - 1 for d in range(k)]
-        for size, min_size, max_size in zip(output_size, min_sizes, max_sizes):
-            if size < min_size or size > max_size:
-                raise ValueError((
-                    "requested an output size of {}, but valid sizes range "
-                    "from {} to {} (for an input of {})").format(
-                        output_size, min_sizes, max_sizes, input.size()[2:]))
-
-        return tuple([output_size[d] - min_sizes[d] for d in range(k)])
+        # 2*padding - output_padding = (input - 1)*stride - output + kernel_size
+        #                            = no_pad_diff
+        # padding = math.ceil(no_pad_diff / 2)
+        def no_pad_diff(d):
+            return ((input.size(d + 2) - 1) * self.stride[d] - output_size[d] + self.kernel_size[d])
+        padding = tuple([(no_pad_diff(d) + 1) // 2 for d in range(k)])
```

Comment:
That's right. It's not compatible to the old version.
The original version tries to keep the self.padding and pad the output on one side. 
However, this version pads the output symmetrically on both sides.
You may just modify your code as follows and print the output and look at the values on the boundaries.
```python
m = nn.ConvTranspose2d(3, 4, 3, 3)
i = Variable(torch.randn(2, 3, 6, 6))
o = m(i, output_size=(20, 20))
print(o)
```

I wonder what would be a better practice.
From my point of view, ConvTranspose2d is wildly used in GANs or AutoEncoders, so padding outputs symmetrically when generating images is more reasonable.

If you think compatibility is more important, then there two ways to adjust this:
1. Try to use self.padding first. If it is impossible to find such output_padding that matches the output_size, then we use the old method. Otherwise, we use the new one.
2. Find a new padding that is the closest to self.padding such that there exists a output_padding (between 0 and self.stride-1) matches the output_size.